### PR TITLE
feat: add editor mode switch 

### PR DESCRIPTION
**Pull request by gorgeousvlad**

### CODE REVIEW

File: src/editor/components/AddBlock/AddBlock.scss [hunk 1]
```diff
@@ -9,6 +9,13 @@ $block: '.#{$ns}add-block';
     $buttonIconSize: 16px;
     $buttonWidth: 76px;
     $buttonHeight: 40px;
+    $zIndex: 110;
```
Comment (NikitaCG) [inline src/editor/components/AddBlock/AddBlock.scss:12]: up to you, but if we use this variable one time, maybe we dont need this variable ?

File: src/editor/components/ControlPanel/ControlPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    viewMode?: ViewModeItem;
+    onViewModeChange: (viewMode: ViewModeItem) => void;
+}
+
+const ControlPanel = ({
+    viewMode = ViewModeItem.Edititng,
+    onViewModeChange,
+    className,
+}: ControlPanelProps) => {
+    return (
```
Comment (NikitaCG) [inline src/editor/components/ControlPanel/ControlPanel.tsx:33]: return jsx immediately

File: src/editor/components/Layout/Layout.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    onModeChange: (mode: ViewModeItem) => void;
+}
+
+const Layout = ({children, mode, onModeChange}: PropsWithChildren<LayoutProps>) => {
+    let left, right;
+
+    function handleChild(child: ReactElement) {
+        switch (child?.type) {
+            case Left:
+                left = child?.props.children;
```
Comment (NikitaCG) [inline src/editor/components/Layout/Layout.tsx:25]: won't we have the same potential problem as recently on the blog? when the structure changes and everything falls apart

File: src/editor/containers/Editor/Editor.tsx [hunk 4]
```diff
@@ -1,70 +1,77 @@
...
-        }),
-        [injectEditBlockProps, content, errorBoundaryState, rest.custom],
-    );
-    const formSpecs = useFormSpec(customSchema);
+    const isEditingMode = viewMode === ViewModeItem.Edititng;
+    const constructorProps = useMemo(() => {
+        if (isEditingMode) {
+            return {
+                content,
+                custom: addCustomDecorator(
```
Comment (NikitaCG) [inline src/editor/containers/Editor/Editor.tsx:31]: maybe refactor like this: 
```
{
  ....
  const customProp = isEditingMode ? ... : rest.custom;
  ...
  return {content, custom: customProp};
}
```

**Review by NikitaCG — CHANGES_REQUESTED** [review]
see in comments

File: src/editor/components/Layout/Layout.tsx (hunk 3 above)
Comment (gorgeousvlad) [reply src/editor/components/Layout/Layout.tsx:25]: it is just a react pattern, we have it in components like CardBase or in cloud site.  It is like Radio and Radio.Button& Layout relies on its own subcomponents structure only and fetch children from its subcomponents, I think it is okay this way)

**Review by yuberdysheva — DISMISSED** [review]

**Review by NikitaCG — APPROVED** [review]

**Review by yuberdysheva — APPROVED** [review]

**Pull request merged**